packages/react-native-windows-init/src/GeneratorShim.ts

```diff
@@ -0,0 +1,177 @@
+    describe:
+      '[Experimental] change to start consuming a NuGet containing a pre-built dll version of Microsoft.ReactNative',
+    hidden: true,
+    default: false,
+  },
+  useHermes: {
+    type: 'boolean' as 'boolean',
+    describe:
+      '[Experimental] Use Hermes instead of Chakra as the JS engine (supported on 0.64+ for C++ projects)',
+    default: false,
+  },
+  useWinUI3: {
+    type: 'boolean' as 'boolean',
+    describe: '[Experimental] Use WinUI3',
+    hidden: true,
+    default: false,
+  },
+  nuGetTestVersion: {
+    type: 'string' as 'string',
+    describe:
+      '[internalTesting] By default the NuGet version matches the rnw package. This flag allows manually specifying the version for internal testing.',
+    hidden: true,
+  },
+  nuGetTestFeed: {
+    type: 'string' as 'string',
+    describe:
+      '[internalTesting] Allows a test feed to be added to the generated NuGet configuration',
+    hidden: true,
+  },
+};
+
+type LegacyCliOptions = yargs.InferredOptionTypes<typeof legacyCliOptions>;
+
+export function registerCliOptions(yargv: yargs.Argv) {
+  const registrationSucceeded = tryRequireChain([
+    () => {
+      // 0.66+
+      const generator = require(require.resolve(
+        'react-native-windows/generator',
+        {
+          paths: [process.cwd()],
+        },
+      ));
+      generator.registerCliOptions(yargv);
+      return true;
+    },
+  ]);
+
+  if (!registrationSucceeded) {
+    yargv.options(legacyCliOptions);
+  }
+}
+
+/**
+ * Returns the Windows template generator, normalizing to any differences
+ * across RN versions.
+ */
+export function requireGenerateProject(): GenerateProject | null {
+  return tryRequireChain<GenerateProject>([
+    () => {
+      const generator = require(require.resolve(
+        'react-native-windows/generator',
+        {
+          paths: [process.cwd()],
+        },
+      ));
+      return generator.generateProject;
+    },
+    () => {
+      const generateProject65: GenerateProject65 = require(require.resolve(
+        'react-native-windows/generate',
+        {
+          paths: [process.cwd()],
+        },
+      ));
+
+      const argv = (yargs.argv as unknown) as LegacyCliOptions;
+      return (projectDir, name, ns, initOptions) =>
+        generateProject65(projectDir, name, ns, {...argv, ...initOptions});
+    },
+    () => {
+      const generateProject62: GenerateProject62 = require(require.resolve(
+        'react-native-windows/local-cli/generate-windows',
+        {paths: [process.cwd()]},
+      ));
+
+      const argv = (yargs.argv as unknown) as LegacyCliOptions;
+      return async (projectDir, name, ns, initOptions) =>
+        generateProject62(projectDir, name, ns, {...argv, ...initOptions});
+    },
+  ]);
+}
+
+/**
+ * Executes each attempt to require in order, returning the result of the
+ * first function to not throw 'MODULE_NOT_FOUND'
+ */
+function tryRequireChain<T>(attempts: Array<() => T>): T | null {
```

Comment:
Wouldn't this make more sense to just check for version? @mganandraj needs that feature too for hermes...
I don't see a difference between the first and second(65) `require` calls... so if the first fails, 65 will fail too not?

Reply:
I was looking into this, and realized there's a more fundamental issue I missed. The generator is only present in the CLI after we have already done quite a bit of work/logic to pick out and install RNW. So, you wouldn't be able to do things like pull into generator commands until the specific version of RNW is resolved.

Thinking about how else we could organize this. I want to figure out how we can get CLI options for a specific RNW version without having to install it first. I've been thinking maybe a third package for just generator args that provides every version (think IFoo1, IFoo2 style).